azure-iot-device/tests/iothub/auth/test_base_renewable_token_authentication_provider.py

```diff
@@ -49,16 +49,12 @@ def parse(source):
 
 @pytest.fixture(scope="function")
 def device_auth_provider():
-    auth_provider = FakeAuthProvider(fake_hostname, fake_device_id, None)
-    yield auth_provider
-    auth_provider.disconnect()
```

Comment:
>auth_provider.disconnect() [](start = 4, length = 26)

Took a look at the source, and it doesn't look like removing this would cause any problems, but if that's true, was the inclusion of it here unneeded before?

Reply:
disconnect used to be a cleanup function on auth provider.  I added this long ago, before I knew that daemon threads were a thing.  now, instead of having an explicit cleanup via disconnect, I have an implicit cleanup via finalizer. so the disconnect is no longer necessary.

---
In reply to: [332259239](https://github.com/Azure/azure-iot-sdk-python/pull/334#discussion_r332259239) [](ancestors = 332259239)